src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -530,13 +531,9 @@ std::string postQueryContext
   // In API version 2, this has changed completely. Here, the total count of local entities is returned
   // if the URI parameter 'count' is set to 'true', and it is returned in the HTTP header Fiware-Total-Count.
   //
-  std::string georel  =  ciP->uriParam["georel"];
-  if ((ciP->apiVersion == V2))
+  if ((ciP->apiVersion == V2) && (ciP->uriParamOptions["count"]))
   {
-    if (!((ciP->inMimeType == JSON && strstr(ciP->payload, "near")) || strstr(georel.c_str(), "near")))
-    {
-      countP = &count;
-    }
+    countP = &count;
```

Comment:
After this change, it seems it keeps working:

```
$ ./testHarness.sh cases/4114_geojson_feature_support
lun 05 jun 2023 10:00:11 CEST
Run tests 0 to 0
0001/6: geojson_feature_errors.test ..................................................................................  04 seconds
0002/6: geojson_feature_errors_ignoretype.test .......................................................................  04 seconds
0003/6: geojson_feature_support.test .................................................................................  02 seconds
0004/6: geojson_feature_support_ignoretype.test ......................................................................  03 seconds
0005/6: geojson_featurecollection_support.test .......................................................................  02 seconds
0006/6: geojson_featurecollection_support_ignoretype.test ............................................................  03 seconds
Total test time: 18.30 seconds

$ ./testHarness.sh cases/1038_ngsiv2_geo_location_entities/
lun 05 jun 2023 10:00:41 CEST
Run tests 0 to 0
0001/16: entity_location_shape_changes.test ...........................................................................  05 seconds
0002/16: errors_general.test ..........................................................................................  04 seconds
0003/16: errors_geobox.test ...........................................................................................  04 seconds
0004/16: errors_geoline.test ..........................................................................................  05 seconds
0005/16: errors_geopoint.test .........................................................................................  03 seconds
0006/16: errors_geopolygon.test .......................................................................................  04 seconds
0007/16: geoquery_box_geobox.test .....................................................................................  03 seconds
0008/16: geoquery_box_geobox_after_update.test ........................................................................  03 seconds
0009/16: geoquery_box_geoline.test ....................................................................................  03 seconds
0010/16: geoquery_box_geoline_after_update.test .......................................................................  03 seconds
0011/16: geoquery_box_geopolygon.test .................................................................................  02 seconds
0012/16: geoquery_box_geopolygon_after_update.test ....................................................................  03 seconds
0013/16: geoquery_circle_geopoint.test ................................................................................  05 seconds
0014/16: geoquery_circle_geopoint_after_update.test ...................................................................  04 seconds
0015/16: geoquery_polygon_geopoint.test ...............................................................................  06 seconds
0016/16: geoquery_polygon_geopoint_after_update.test ..................................................................  06 seconds
Total test time: 62.86 seconds

$ ./testHarness.sh cases/1177_geometry_and_coords/
lun 05 jun 2023 10:02:29 CEST
Run tests 0 to 0
0001/2: geometry_and_coords.test .....................................................................................  03 seconds
0002/2: geometry_and_coords_errors.test ..............................................................................  03 seconds
Total test time: 6.20 seconds
```

Reply:
NTC (informative)

Reply:
After the above changes, The FTs (1038, 1177, 4114) will work fine. I also have same understanding. 

But, with above code changes, The Pagination functionality (in case of Request Forwarding to CPr) will not work. Because, as I've mentioned in [this comment](https://github.com/telefonicaid/fiware-orion/pull/4341#issuecomment-1566149834), I need the value of `countP` variable to calculate the offset properly at L#725 in modified postQueryContext.cpp file. 

![offset_calculation_logic_PR#4275](https://github.com/telefonicaid/fiware-orion/assets/55977298/833a81da-e5ed-4ee7-ab99-20c4b51474f7)


I have executed the `4149_pagination_for_request_forwarding/pagination_for_request_forwarding.test` FT file with your modification but this FT is getting failed. Please find below screenshot for the same,

![4149_pagination FT_failed](https://github.com/telefonicaid/fiware-orion/assets/55977298/8c22b4b2-12bf-4480-acf0-cb16c1888acc)

It is happening because of below changes i.e.
```
-  if ((ciP->apiVersion == V2))
+  if ((ciP->apiVersion == V2) && (ciP->uriParamOptions["count"]))
```

Due to this the `countP` pointer variable remain `NULL` in case of below API request (as `options=count` is not passed in the request):
```
curl -X GET 'localhost:1026/v2/entities?idPattern=.*&type=T&limit=10&offset=10'
or 
curl -X GET 'localhost:1026/v2/entities?idPattern=.*&type=T&limit=10'
```

In these cases, Broker is getting exited at L#725 of modified `postQueryContext.cpp` file.

Reply:
Thank you for the detailed feedback! I'll look it in detail.

Reply:
I have re-enabled the logic in commit [c42a534](https://github.com/telefonicaid/fiware-orion/pull/4352/commits/c42a534b116d62dcd8d464bd4fb4222d38881f9e)

This is going to make FT (1038, 1177, 4114) to fail, but I think is better to merge this PR as it is (as it also has the skip NGSIv1 logic and the documentation) and then continue to work on it.